src/main/java/org/wildfly/installationmanager/spi/InstallationManager.java

```diff
@@ -119,4 +120,22 @@ public interface InstallationManager {
      * @throws Exception
      */
     Path createSnapshot(Path targetPath) throws Exception;
+
+    /**
+     * Generate an apply update CLI command.
+     * The generated command can be run in separate process to apply changes.
+     *
+     * @param candidatePath - path to the update candidate
+     * @return empty Optional if the CLI is not available, or a CLI command.
+     */
+    Optional<String> generateApplyUpdateCommand(Path candidatePath);
```

Comment:
@spyrkob What about --dir? The SPI is not getting this value, how it would generate the --dir argument?

Reply:
It takes it from the `installationDir` in `InstallationManagerFactory`. 

The SPI is meant to works on a single installation, so the apply operations use the base server as target.

Reply:
Ahh, ok, got it, via Maven Options, as the other actions implemented, I forgot about it.